emcc.py

```diff
@@ -1603,6 +1603,11 @@ def save_intermediate(name=None, suffix='js'):
         link_opts = [] if debug_level >= 4 or shared.Settings.CYBERDWARF else ['-strip-debug'] # remove LLVM debug if we are not asked for it
         if not shared.Settings.ASSERTIONS:
           link_opts += ['-disable-verify']
+        else:
+          # when verifying, LLVM debug info has some tricky linking aspects, and llvm-link will
+          # disable the type map in that case. we added linking to opt, so we need to do
+          # something similar, which we can do with a param to opt
+          link_opts += ['-disable-debug-info-type-map']
```

Comment:
Yeah, we can link in the opt tool - as mentioned in that comment and above in the source, we can provide multiple inputs to opt, and it links them (we modified opt that way). This saves time by not running llvm-link separately.

Bascially, turns out that linking is a little tricky in a weird way, and llvm-link handled that, so we just need to do the same thing when we link in opt.